DependencyInjection/DoctrineExtension.php

```diff
@@ -117,8 +117,22 @@ protected function dbalLoad(array $config, ContainerBuilder $container)
      */
     protected function loadDbalConnection($name, array $connection, ContainerBuilder $container)
     {
-        $configuration = $container->setDefinition(sprintf('doctrine.dbal.%s_connection.configuration', $name), new ChildDefinition('doctrine.dbal.connection.configuration'));
-        $logger        = null;
+        $configuration       = $container->setDefinition(sprintf('doctrine.dbal.%s_connection.configuration', $name), new ChildDefinition('doctrine.dbal.connection.configuration'));
+        $connectionServiceId = sprintf('doctrine.dbal.%s_connection', $name);
+
+        if (isset($connection['factory'])) {
+            $definition = new Definition();
+            $definition
+                ->setPublic(true)
+                ->setFactory(new Reference($connection['factory']))
+                ->setArguments($connection['factory_arguments']);
+
+            $container->setDefinition($connectionServiceId, $definition);
+
+            return;
```

Comment:
This is why I hesitated to accept this feature request. Custom factory on config level like this means ignoring all the other config options. This is not desirable. It's not expected by user that when they see config of connection1 composed of 10 keys, only 1 is applied and rest are ignored. If this feature is accepted, I think it should either:
- Not be part of the config, only generate `doctrine.dbal.%_connection.factory` definitions for each connection - then users can just redefine it. This makes it obvious that since extension point is on DI level, fault cannot be in doctrine-bundle config. But I actually don't see why can't you just define your `doctrine.dbal.%_connection` service, this is almost same thing.
- Be part of a config, but all existing config options need to be in effect. That would probably mean there needs to be prescribed certain format of factory which would be compatible with our existing factory - so that all values that are generated here can be passed to it.

Reply:
🤦  for reasons which will forever remain a mystery, I was under the impression that to create my own connection, I have to first create it in the `dbal` section and only then override it. But that's of course not the case and it works.
Thanks for your help! Feel free to close this PR.